roscala/src/test/scala/coop/rchain/rosette/TransitionSpec.scala

```diff
@@ -524,4 +524,61 @@ class TransitionSpec extends FlatSpec with Matchers {
     end.ctxt.ctxt.trgt shouldBe Fixnum(100)
   }
 
+  "OpRtnArg" should "copy ctxt.rslt to given argvec element" in {
+
+    /**
+      * This opcode sequence will never be generated by the RBL
+      * compiler as it is the responsibility of the caller to
+      * specify where the result will be stored.
+      *
+      * litvec:
+      *   0:   100
+      * codevec:
+      *   0:   liti 0,rslt
+      *   1:   rtn/nxt arg[1]
+      */
+    val start =
+      testState
+        .set(_ >> 'ctxt >> 'ctxt)(testState.ctxt)
+        .set(_ >> 'ctxt >> 'argvec)(Tuple(Seq(Ob.NIV, Ob.NIV, Ob.NIV)))
+        .update(_ >> 'code >> 'litvec)(_ => Tuple(Seq(Fixnum(100))))
+
+    val codevec = Seq(
+      OpIndLitToRslt(lit = 0),
+      OpRtnArg(next = true, arg = 1)
+    )
+
+    val end = VirtualMachine.executeSeq(codevec, start)
+    end.ctxt.ctxt.argvec.elem(1) shouldBe Fixnum(100)
+  }
+
+  "OpRtnTag" should "copy ctxt.rslt to given location" in {
+
+    /**
+      * This opcode sequence will never be generated by the RBL
+      * compiler as it is the responsibility of the caller to
+      * specify where the result will be stored.
+      *
+      * litvec:
+      *   0:   100
+      *   1:   {Loc arg[2]}
+      * codevec:
+      *   0:   liti 0,rslt
+      *   1:   rtn/nxt
+      */
+    val start =
+      testState
+        .set(_ >> 'ctxt >> 'ctxt)(testState.ctxt)
+        .set(_ >> 'ctxt >> 'argvec)(Tuple(Seq(Ob.NIV, Ob.NIV, Ob.NIV)))
+        .update(_ >> 'code >> 'litvec)(_ =>
+          Tuple(Seq(Fixnum(100), ArgRegister(1))))
```

Comment:
In the description `Loc arg[2]` is specified, but here `ArgRegister(1)` is used. Did I miss something?

Reply:
Good catch.